### ISSUE TITLE
Using "remove" option might introduce untransliterated symbols

### ISSUE THREAD

**Issue by x-yuri** [simov/slugify#65]
```js
const slugify = require('slugify');
console.log(slugify('Аленький цветочек'));  // Alenkij-cvetochek
console.log(slugify('Аленький цветочек', {remove: /[()]+/g}));  // Alenьkij-cvetochek
```

That happens because `slugify` first removes the symbol, but then right away adds it [back][1]:

```js
(charMap[ch] || ch).replace(options.remove || /[^\w\s$*_+~.()'"!\-:@]/g, '')
charMap['ь'] == ''
charMap['ь'] || 'ь' == 'ь'
```

[1]: https://github.com/simov/slugify/blob/v1.3.5/slugify.js#L31

**Comment by simov** [comment]
Good point, any idea what symbol should we use to replace `ь` by default?

**Comment by x-yuri** [comment]
Some sort of [apostrophe or nothing][1]. "ь" basically makes the preceding sound [softer][1]. Is there a reason not to let letters transliterate to nothing?

By the way, "ъ" becomes "u"? That's one strange idea.

On a side note, I wouldn't make the tranliteration table one long string.

[1]: https://en.wikipedia.org/wiki/Romanization_of_Russian#Transliteration_table
[2]: https://en.wikipedia.org/wiki/Soft_sign

**Comment by simov** [comment]
It is [nothing](https://github.com/simov/slugify/blob/master/config/charmap.json#L286) currently, that's why it fails. I have to probably make the condition a bit smarter.

`ъ` comes from [Bulgarian](https://en.wikipedia.org/wiki/Romanization_of_Bulgarian) though not sure why in case the Phonetic Keyboard layout is using `y` for `ъ` and `u` for `у` .. anyway, note that many symbols end up being used in multiple languages and have different phonetics in some. That's why either some sort of locale have to be implemented in this module, or you have to use the `extend` method to set the characters that do not match your language.

**Comment by filiphanes** [comment]
same behavior is for `slugify("otázky", {remove: /x/g})` => `"otázky"`

**Comment by x-yuri** [comment]
Technically, that's a different issue, although the effect is the same (not that I'm suggesting to create another issue).

`slugify` for each character does:

```js
        return result + (locale[ch] || charMap[ch] || ch)
          // remove not allowed characters
          .replace(options.remove || /[^\w\s$*_+~.()'"!\-:@]+/g, '')
```

Neither I, nor you specify a locale, so the `locale[ch]` part can be ignored. But in my case (`ch = 'ь'`) `charMap[ch] == ''`, `charMap[ch] || ch == 'ь'`, and then `ь` is removed by the default regex. When I specify `options.remove` that no longer happens.

In your case `á` is `LATIN SMALL LETTER A (U+0061)` + `COMBINING ACUTE ACCENT (U+0301)`. `U+0061` always remains, since it's a legit English letter. `U+0301` is replaced by the default regex. Unless you pass `options.remove`.
